### PR TITLE
toxcore: update to 0.2.7.

### PR DESCRIPTION
**Pull request by iphydf**

### CODE REVIEW

File: chat/toxcore/patches/patch-testing_misc__tools.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+$NetBSD$
+
+--- testing/misc_tools.c.orig	2018-08-16 20:45:33.000000000 +0000
++++ testing/misc_tools.c
+@@ -27,7 +27,7 @@
+ 
+ #ifndef _POSIX_C_SOURCE
+ // For nanosleep().
+-#define _POSIX_C_SOURCE 199309L
++#define _POSIX_C_SOURCE 200112L
```
Comment (krytarowski) [inline chat/toxcore/patches/patch-testing_misc__tools.c]: Why this?
If we want to overwrite or define `_POSIX_C_SOURCE` we shall do it from main package's level with `-D_POSIX_C_SOURCE=200112L`

**Comment by krytarowski** [comment]
Please add proposed changelog entry.

**Comment by iphydf** [comment]
Where do I add the changelog entry?

File: chat/toxcore/patches/patch-testing_misc__tools.c (hunk 1 above)
Comment (iphydf) [reply chat/toxcore/patches/patch-testing_misc__tools.c]: Done.

**Comment by krytarowski** [comment]
Here so a committer can land this change with a ready changelog.

**Comment by iphydf** [comment]
* Fix problems with initial connections and name-setting in conferences
* Add callback for successful connection to a conference
* Fix memory leak in error path in group A/V

**Comment by iphydf** [comment]
Any action required?

**Comment by krytarowski** [comment]
It's pending until someone else than me will grab it, I'm swamped to do it myself (and I don't know how to test it myself).

**Comment by iamleot** [comment]
Hello iphydf,
First of all thank you very much for updating it!

iphydf writes:
> Any action required?

Probably it's worth to first import it in pkgsrc-wip to ease the
review process (ATM, at least on NetBSD-current the build seems to
fail, full log at the end of this reply).

If you do not have access to pkgsrc-wip please give a look to:

 <https://www.pkgsrc.org/wip/users/>

On a (not-so)-current NetBSD the build fails with:

    [...]
    [ 53%] Built target auto_typing_test
    In file included from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:7:0:
    /usr/include/g++/cstdio:176:11: error: '::vfscanf' has not been declared
       using ::vfscanf;
               ^~~~~~~
    Scanning dependencies of target auto_lan_discovery_test
    Scanning dependencies of target auto_encryptsave_test
    [ 53%] Building C object CMakeFiles/auto_lan_discovery_test.dir/auto_tests/lan_discovery_test.c.o
    /usr/include/g++/cstdio:177:11: error: '::vscanf' has not been declared
       using ::vscanf;
               ^~~~~~
    /usr/include/g++/cstdio:179:11: error: '::vsscanf' has not been declared
       using ::vsscanf;
               ^~~~~~~
    /usr/include/g++/cstdio:186:22: error: '__gnu_cxx::vfscanf' has not been declared
       using ::__gnu_cxx::vfscanf;
                          ^~~~~~~
    /usr/include/g++/cstdio:187:22: error: '__gnu_cxx::vscanf' has not been declared
       using ::__gnu_cxx::vscanf;
                          ^~~~~~
    /usr/include/g++/cstdio:189:22: error: '__gnu_cxx::vsscanf' has not been declared
       using ::__gnu_cxx::vsscanf;
                          ^~~~~~~
    In file included from /usr/include/g++/bits/localefwd.h:42:0,
                     from /usr/include/g++/string:43,
                     from /usr/include/g++/stdexcept:39,
                     from /usr/include/g++/array:39,
                     from /usr/include/g++/tuple:39,
                     from /usr/include/g++/functional:55,
                     from /usr/include/g++/memory:79,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:9:
    /usr/include/g++/cctype:87:11: error: '::isblank' has not been declared
       using ::isblank;
               ^~~~~~~
    [ 54%] Building C object CMakeFiles/auto_encryptsave_test.dir/auto_tests/encryptsave_test.c.o
    [ 55%] Linking C executable tox-bootstrapd
    [ 55%] Built target tox-bootstrapd
    [ 56%] Linking C executable auto_lan_discovery_test
    Scanning dependencies of target auto_toxav_basic_test
    [ 57%] Building C object CMakeFiles/auto_toxav_basic_test.dir/auto_tests/toxav_basic_test.c.o
    In file included from /usr/include/g++/random:38:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/cmath:1101:11: error: '::double_t' has not been declared
       using ::double_t;
               ^~~~~~~~
    [ 57%] Built target auto_lan_discovery_test
    /usr/include/g++/cmath:1102:11: error: '::float_t' has not been declared
       using ::float_t;
               ^~~~~~~
    /usr/include/g++/cmath:1105:11: error: '::acosh' has not been declared
       using ::acosh;
               ^~~~~
    /usr/include/g++/cmath:1109:11: error: '::asinh' has not been declared
       using ::asinh;
               ^~~~~
    /usr/include/g++/cmath:1113:11: error: '::atanh' has not been declared
       using ::atanh;
               ^~~~~
    /usr/include/g++/cmath:1117:11: error: '::cbrt' has not been declared
       using ::cbrt;
               ^~~~
    /usr/include/g++/cmath:1121:11: error: '::copysign' has not been declared
       using ::copysign;
               ^~~~~~~~
    /usr/include/g++/cmath:1125:11: error: '::erf' has not been declared
       using ::erf;
               ^~~
    /usr/include/g++/cmath:1129:11: error: '::erfc' has not been declared
       using ::erfc;
               ^~~~
    /usr/include/g++/cmath:1137:11: error: '::expm1' has not been declared
       using ::expm1;
               ^~~~~
    /usr/include/g++/cmath:1157:11: error: '::hypot' has not been declared
       using ::hypot;
               ^~~~~
    /usr/include/g++/cmath:1161:11: error: '::ilogb' has not been declared
       using ::ilogb;
               ^~~~~
    /usr/include/g++/cmath:1165:11: error: '::lgamma' has not been declared
       using ::lgamma;
               ^~~~~~
    /usr/include/g++/cmath:1177:11: error: '::log1p' has not been declared
       using ::log1p;
               ^~~~~
    /usr/include/g++/cmath:1185:11: error: '::logb' has not been declared
       using ::logb;
               ^~~~
    /usr/include/g++/cmath:1205:11: error: '::nextafter' has not been declared
       using ::nextafter;
               ^~~~~~~~~
    /usr/include/g++/cmath:1213:11: error: '::remainder' has not been declared
       using ::remainder;
               ^~~~~~~~~
    /usr/include/g++/cmath:1221:11: error: '::rint' has not been declared
       using ::rint;
               ^~~~
    /usr/include/g++/cmath:1229:11: error: '::scalbln' has not been declared
       using ::scalbln;
               ^~~~~~~
    /usr/include/g++/cmath:1233:11: error: '::scalbn' has not been declared
       using ::scalbn;
               ^~~~~~
    Scanning dependencies of target auto_crypto_test
    [ 58%] Building C object CMakeFiles/auto_crypto_test.dir/auto_tests/crypto_test.c.o
    [ 58%] Linking C executable auto_encryptsave_test
    In file included from /usr/include/g++/random:51:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/bits/random.tcc: In member function 'void std::poisson_distribution<_IntType>::param_type::_M_initialize()':
    /usr/include/g++/bits/random.tcc:1250:32: error: call of overloaded 'lgamma(double)' is ambiguous
        _M_lfm = std::lgamma(__m + 1);
                                    ^
    In file included from /usr/include/g++/random:38:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/cmath:1531:3: note: candidate: constexpr float std::lgamma(float)
       lgamma(float __x)
       ^~~~~~
    /usr/include/g++/cmath:1535:3: note: candidate: constexpr long double std::lgamma(long double)
       lgamma(long double __x)
       ^~~~~~
    In file included from /usr/include/g++/random:51:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/bits/random.tcc: In member function 'std::poisson_distribution<_IntType>::result_type std::poisson_distribution<_IntType>::operator()(_UniformRandomNumberGenerator&, const std::poisson_distribution<_IntType>::param_type&)':
    /usr/include/g++/bits/random.tcc:1357:52: error: call of overloaded 'lgamma(double)' is ambiguous
            > __param._M_lfm - std::lgamma(__x + __m + 1));
                                                        ^
    In file included from /usr/include/g++/random:38:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/cmath:1531:3: note: candidate: constexpr float std::lgamma(float)
       lgamma(float __x)
       ^~~~~~
    /usr/include/g++/cmath:1535:3: note: candidate: constexpr long double std::lgamma(long double)
       lgamma(long double __x)
       ^~~~~~
    In file included from /usr/include/g++/random:51:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/bits/random.tcc: In member function 'void std::binomial_distribution<_IntType>::param_type::_M_initialize()':
    /usr/include/g++/bits/random.tcc:1484:33: error: call of overloaded 'lgamma(double)' is ambiguous
        _M_lf = (std::lgamma(__np + 1)
                                     ^
    In file included from /usr/include/g++/random:38:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/cmath:1531:3: note: candidate: constexpr float std::lgamma(float)
       lgamma(float __x)
       ^~~~~~
    /usr/include/g++/cmath:1535:3: note: candidate: constexpr long double std::lgamma(long double)
       lgamma(long double __x)
       ^~~~~~
    In file included from /usr/include/g++/random:51:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/bits/random.tcc: In member function 'std::binomial_distribution<_IntType>::result_type std::binomial_distribution<_IntType>::operator()(_UniformRandomNumberGenerator&, const std::binomial_distribution<_IntType>::param_type&)':
    /usr/include/g++/bits/random.tcc:1624:35: error: call of overloaded 'lgamma(double)' is ambiguous
             std::lgamma(__np + __x + 1)
                                       ^
    In file included from /usr/include/g++/random:38:0,
                     from /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.6/testing/random_testing.cc:10:
    /usr/include/g++/cmath:1531:3: note: candidate: constexpr float std::lgamma(float)
       lgamma(float __x)
       ^~~~~~
    /usr/include/g++/cmath:1535:3: note: candidate: constexpr long double std::lgamma(long double)
       lgamma(long double __x)
       ^~~~~~
    [...]


Thank you!

**Comment by krytarowski** [comment]
vscanf(3) is protected with `defined(_ISOC99_SOURCE) || defined(_NETBSD_SOURCE)`.. aren't we defining the former for c99?

**Comment by iphydf** [comment]
@iamleot is this a regression over 0.2.5?

**Comment by iamleot** [comment]
Hello iphydf,

iphydf writes:
> @iamleot is this a regression over 0.2.5?

Yes, toxcore-0.2.5 built fine.

**Comment by iphydf** [comment]
That's very strange.

I'm doing `git diff v0.2.5..v0.2.6` on several files that could be involved in this: CMakeLists.txt, the cmake directory, and random_testing.cc (which starts with system includes, so other files in the repo can't affect this error). Maybe the `-D` thing was the problem. In 0.2.7 (released today), that will be gone again, so let's try this.

**Comment by iphydf** [comment]
Does 0.2.7 build fine again?

**Comment by iamleot** [comment]
Leonardo Taccari writes:
> [...]
> Yes, toxcore-0.2.5 built fine.
> [...]

Whoops, sorry, I meant toxcore-0.2.4 (latest version in pkgsrc-current)!

**Comment by iamleot** [comment]
Hello iphydf,

iphydf writes:
> Does 0.2.7 build fine again?

Yes, now it builds and install fine, thanks!

However - but please note that this is not a regression! - the test
seems stuck, i.e. doing a `make test':

    % make test
    => Bootstrap dependency digest>=20010302: found digest-20160304
    ===> Testing for toxcore-0.2.7
    Running tests...
    Test project /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.7
          Start  1: TCP

...and nothing seems to progress (doing a ^T I can see
`load: 0.00  cmd: make 27707 [wait] 0.00u 0.01s 0% 2128k').

Can you reproduce that as well and/or any idea on what's going on?

(Of course for any possible further testing and/or more debugging
output do not hesitate to contact me!)

**Comment by iphydf** [comment]
```
Test project /home/pbulk/build/chat/toxcore/work/c-toxcore-0.2.7
      Start  1: TCP
 1/41 Test  #1: TCP ..............................   Passed   15.93 sec
```

It works for me on SmartOS. Can you attach gdb and see where it's stuck? 0% cpu could mean it's stuck in a system call (e.g. networking).

**Comment by iamleot** [comment]
iphydf writes:
> ```
> Test project /home/pbulk/build/chat/toxcore/work/c-toxcore-0.2.7
>       Start  1: TCP
>  1/41 Test  #1: TCP ..............................   Passed   15.93 sec
> ```
>
> It works for me on SmartOS. Can you attach gdb and see where it's stuck? 0% cpu could mean it's stuck in a system call (e.g. networking).
>

Sure (I have recompiled it via `CFLAGS=-g3'):

    % gdb -p `pgrep auto_TCP_test`
    [...]
    (gdb) bt
    #0  0x000073c4856d630a in recvfrom () from /usr/lib/libc.so.12
    #1  0x0000000000405dce in net_recv (sock=..., sock@entry=..., buf=buf@entry=0x7f7fffc6f720, len=len@entry=96)
        at /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.7/toxcore/network.c:1440
    #2  0x000000000040358a in test_basic () at /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.7/auto_tests/TCP_test.c:114
    #3  0x000000000042a66b in TCP_suite () at /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.7/auto_tests/TCP_test.c:844
    #4  main () at /tmp/pkgsrc/chat/toxcore/work/c-toxcore-0.2.7/auto_tests/TCP_test.c:857


Thanks!

**Comment by iphydf** [comment]
I'll see if I can get a netbsd VM up and running to test this.

Filed https://github.com/TokTok/c-toxcore/issues/1146. Do any of the other tests work? E.g. if you do ./auto_conference_test?

**Comment by iamleot** [comment]
Hello iphydf,

iphydf writes:
> I'll see if I can get a netbsd VM up and running to test this.
>
> Filed https://github.com/TokTok/c-toxcore/issues/1146. Do any of the other tests work? E.g. if you do ./auto_conference_test?
>

Here the complete output:

 | % ./auto_conference_test
 | creating 16 toxes
 | [#1] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#1] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#1] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33445
 | [#2] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#2] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#2] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33446
 | [#3] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#3] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#3] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33447
 | [#4] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#4] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#4] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33448
 | [#5] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#5] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#5] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33449
 | [#6] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#6] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#6] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33450
 | [#7] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#7] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#7] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33451
 | [#8] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#8] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#8] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33452
 | [#9] DEBUG network.c:836        new_networking_ex:      Dual-stack socket: enabled
 | [#9] DEBUG network.c:849        new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#9] DEBUG network.c:881        new_networking_ex:      Bound successfully to [::]:33453
 | [#10] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#10] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#10] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33454
 | [#11] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#11] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#11] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33455
 | [#12] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#12] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#12] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33456
 | [#13] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#13] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#13] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33457
 | [#14] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#14] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#14] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33458
 | [#15] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#15] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#15] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33459
 | [#16] DEBUG network.c:836       new_networking_ex:      Dual-stack socket: enabled
 | [#16] DEBUG network.c:849       new_networking_ex:      Failed to activate local multicast membership. (49, Can't assign requested address)
 | [#16] DEBUG network.c:881       new_networking_ex:      Bound successfully to [::]:33460
 | creating a chain of friends
 | waiting for everyone to come online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | tox #11: is now connected
 | currently 0 toxes are online
 | tox #1: is now connected
 | tox #2: is now connected
 | tox #5: is now connected
 | tox #10: is now connected
 | tox #12: is now connected
 | tox #13: is now connected
 | tox #16: is now connected
 | currently 0 toxes are online
 | tox #3: is now connected
 | tox #7: is now connected
 | tox #14: is now connected
 | tox #15: is now connected
 | currently 0 toxes are online
 | tox #4: is now connected
 | tox #8: is now connected
 | tox #9: is now connected
 | currently 0 toxes are online
 | tox #6: is now connected
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | currently 0 toxes are online
 | tox #13: is now connected to friend 0
 | currently 1 toxes are online
 | tox #5: is now connected to friend 0
 | tox #11: is now connected to friend 0
 | tox #12: is now connected to friend 1
 | tox #14: is now connected to friend 1
 | tox #15: is now connected to friend 0
 | currently 4 toxes are online
 | tox #4: is now connected to friend 1
 | [#4] DEBUG net_crypto.c:1195    send_lossless_packet:   send_data_packet failed
 | tox #10: is now connected to friend 1
 | tox #13: is now connected to friend 1
 | tox #14: is now connected to friend 0
 | currently 5 toxes are online
 | tox #11: is now connected to friend 1
 | tox #12: is now connected to friend 0
 | currently 6 toxes are online
 | tox #1: is now connected to friend 0
 | tox #2: is now connected to friend 0
 | tox #7: is now connected to friend 0
 | tox #8: is now connected to friend 1
 | tox #9: is now connected to friend 1
 | tox #9: is now connected to friend 0
 | tox #10: is now connected to friend 0
 | currently 11 toxes are online
 | tox #6: is now connected to friend 1
 | tox #8: is now connected to friend 0
 | currently 12 toxes are online
 | tox #7: is now connected to friend 1
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | currently 12 toxes are online
 | tox #5: is now connected to friend 1
 | tox #6: is now connected to friend 0
 | currently 13 toxes are online
 | tox #3: is now connected to friend 0
 | currently 14 toxes are online
 | tox #2: is now connected to friend 1
 | tox #16: is now connected to friend 0
 | currently 15 toxes are online
 | tox #15: is now connected to friend 1
 | currently 15 toxes are online
 | tox #3: is now connected to friend 1
 | tox #4: is now connected to friend 0
 | currently 16 toxes are online
 | friends connected, took 59 seconds
 | tox #1: inviting its first friend
 | waiting for invitations to be made
 | tox #2: invited next friend
 | tox #3: invited next friend
 | tox #4: invited next friend
 | tox #5: invited next friend
 | tox #6: invited next friend
 | tox #7: invited next friend
 | tox #8: invited next friend
 | tox #9: invited next friend
 | tox #10: invited next friend
 | tox #11: invited next friend
 | tox #12: invited next friend
 | tox #13: invited next friend
 | tox #14: invited next friend
 | tox #15: invited next friend
 | waiting for all toxes to be in the group
 | current peer counts: [9, 9, 10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16]
 | current peer counts: [9, 10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 16]
 | current peer counts: [10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 16, 16]
 | current peer counts: [[#1] DEBUG net_crypto.c:1195      send_lossless_packet:   send_data_packet failed
 | 10[#2] DEBUG net_crypto.c:1195  send_lossless_packet:   send_data_packet failed
 | , 11[#3] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 11[#4] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 12, 12, 13[#7] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 13, 14, 14, 15, 15, 16, 16, 16, 16, 16]
 | current peer counts: [11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 16, 16, 16, 16]
 | current peer counts: [11, 12, 12, 13[#5] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 13, 14[#7] DEBUG net_crypto.c:1195    send_lossless_packet:   send_data_packet failed
 | , 14, 15, 15[#10] DEBUG net_crypto.c:1195       send_lossless_packet:   send_data_packet failed
 | , 16, 16, 16, 16, 16, 16, 16]
 | current peer counts: [[#1] DEBUG net_crypto.c:1195      send_lossless_packet:   send_data_packet failed
 | 12[#2] DEBUG net_crypto.c:1195  send_lossless_packet:   send_data_packet failed
 | , 12[#3] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 13[#4] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 13[#5] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 14[#6] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 14[#7] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 15[#8] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 15[#9] DEBUG net_crypto.c:1195        send_lossless_packet:   send_data_packet failed
 | , 16[#10] DEBUG net_crypto.c:1195       send_lossless_packet:   send_data_packet failed
 | , 16, 16, 16, 16, 16, 16, 16]
 | current peer counts: [[#1] DEBUG net_crypto.c:1195      send_lossless_packet:   send_data_packet failed
 | 12, 13[#3] DEBUG net_crypto.c:1195      send_lossless_packet:   send_data_packet failed
 | , 15, 14, 15, 15, 15, 16, 16, 16, 16, 16, 16, 16, 16, 16]
 | current peer counts: [13, 16, 15, 15, 15, 15, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16]
 | current peer counts: [16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16]
 | group connected, took 2 seconds
 | [#1] DEBUG net_crypto.c:1195    send_lossless_packet:   send_data_packet failed
 | [#6] DEBUG net_crypto.c:1195    send_lossless_packet:   send_data_packet failed
 | tearing down toxes
 | test_many_group succeeded, took 73 seconds


Thanks!

**Comment by iphydf** [comment]
Thanks, that means toxcore public API works fine, it's just the TCP test.

**Comment by coypoop** [comment]
Thanks for the patch

**Comment by bsiegert** [comment]
@iphydf Can you close this? Thanks!

**Pull request closed without merging**